picard/ui/metadatabox/__init__.py

```diff
@@ -274,8 +274,13 @@ def _copy_value(self):
                 value = self.tag_diff.orig[tag]
             elif column == self.COLUMN_NEW:
                 value = self.tag_diff.new[tag]
+
             if tag == '~length':
-                value = [format_time(value or 0), ]
+                try:
+                    value = [format_time(value or 0), ]
+                except (TypeError, ValueError):
```

Comment:
We should not hide those exceptions totally, especially if we want to change how `~length` works (see below).
At least log a warning in case of exception:


```suggestion
                except (TypeError, ValueError) as why:
                    log.warning(why)
```

Also, we expect `value` to contain a positive integer number (time in milliseconds), so the actual conversion should use `int()` and check it is a positive number (because it will happily convert -1000 without raising any error). This can be done in another PR, and matching tests will have to be added (test_utils.py has tests for `format_time()` already.

To me, `format_time()` should only be called with values that can actually be converted, so we should determine if we have one or more than one item in the selection and act accordingly (call `format_time()` for each value).

Reply:
Will raise a new PR for making `format_time()` expect positive ints